### PR TITLE
Remove remaining directory server mentions

### PR DESCRIPTION
**Pull request by pljones**
**Short description of changes**

The overall goal is that:
- _Directory_ is the Jamulus term for an application that supplies a list of Jamulus _Servers_ on request (most often the Jamulus application running as `--server --directoryaddress localhost` (or `-s -e 127.0.0.1`)).
- Directory _Address_ is the IP address (IPv4 only, currently), possibly including port, used to contact the Directory.
- "Directory server" is an invalid term in Jamulus vocabulary.

This removes remaining "directory server" mentions in the code base (based on `grep -ni directory\ \*server src/*.*`).

Apart from:
- backwards compatibility for `--directoryserver` command line option
- RPC doccomment

(Replaced: https://github.com/jamulussoftware/jamulus/pull/2997 - this is more complete)

CHANGELOG: Rename `--directoryserver` to `--directoryaddress` (and internal changes)

**Context: Fixes an issue?**

On going maintenance to avoid confusion in terminology.

**Does this change need documentation? What needs to be documented and how?**

Yes.  `--help` output now says `--directoryaddress` plus a few other changes.    Translations affected in places, too (I added some more server GUI ones).

**Status of this Pull Request**

It runs okay.

**What is missing until this pull request can be merged?**

- Review.
- Fixing `generate_json_rpc_docs.py` script (@hoffie ?)
- Fixing `clang_format`...

Note: the Json RPC "directory_server" mention is _in its external protocol_, not just a comment, so needs consideration before just changing it.  The other changes here affect nothing outside Jamulus.

## Checklist

-  [x] I've verified that this Pull Request follows the [general code principles](https://github.com/jamulussoftware/jamulus/blob/main/CONTRIBUTING.md#jamulus-projectsource-code-general-principles)
-  [x] I tested my code and it does what I want
-  [x] My code follows the [style guide](https://github.com/jamulussoftware/jamulus/blob/main/CONTRIBUTING.md#source-code-consistency) 
-  [x] I waited some time after this Pull Request was opened and all GitHub checks completed without errors. 
-  [x] I've filled all the content above

### CODE REVIEW

File: src/connectdlg.cpp [hunk 1]
```diff
@@ -339,8 +336,7 @@ void CConnectDlg::SetServerList ( const CHostAddress& InetAddr, const CVector<CS
     for ( int iIdx = 0; iIdx < iServerInfoLen; iIdx++ )
     {
         // get the host address, note that for the very first entry which is
-        // the directory server, we have to use the receive host address
-        // instead
+        // the directory, we have to use the receive host address instead
```
Comment (pljones) [inline src/connectdlg.cpp:339]:
```suggestion
        // the directory, we have to use the received host address instead
```

**Comment by pljones** [comment]
Huh?
image

And Github `clang_format` and Ubuntu seem to have got out of alignment on my machine again..!

**Comment by pljones** [comment]
So
```
"--clang-format-executable" "/clang-format/clang-format10"
```
```
$ clang-format --version
Ubuntu clang-format version 14.0.0-1ubuntu1
```
I guess I need to find `clang-format-10` again...

**Comment by pljones** [comment]
Well, so clang-format-10 is no longer available and installing 11 made no difference locally.
```
$ sudo apt install clang-format<TAB><TAB>
clang-format     clang-format-11  clang-format-12  clang-format-13  clang-format-14  clang-format-15
```

File: src/serverrpc.cpp [hunk 2]
```diff
@@ -106,7 +106,7 @@ CServerRpc::CServerRpc ( CServer* pServer, CRpcServer* pRpcServer, QObject* pare
     /// @result {string} result.city - The server city.
     /// @result {number} result.countryId - The server country ID (see QLocale::Country).
     /// @result {string} result.welcomeMessage - The server welcome message.
-    /// @result {string} result.directoryServer - The directory server to which this server requested registration, or blank if none.
+    /// @result {string} result.directoryServer - The directory to which this server requested registration, or blank if none.
```
Comment (ann0see) [inline src/serverrpc.cpp:109]: Did you run the Python script to generate the md file or did you edit it manually?

**Comment by ann0see** [comment]
Would `--directory` instead of `--directoryaddress` be confusing?

File: src/serverrpc.cpp (hunk 2 above)
Comment (pljones) [reply src/serverrpc.cpp:109]: Oops, this wasn't meant to change.  It's meant to be covered by the "TODO" about the doccomment.

Comment (ann0see) [reply src/serverrpc.cpp:109]: That's probably why the check fails

**Comment by pljones** [comment]
> Would `--directory` instead of `--directoryaddress` be confusing?

I had `--directory` until this afternoon.  But it's the _address_ (in Internet Address including Port terms) of the Directory, not the Directory itself.  See "The overall goal is that:" in the PR description.

File: src/util.h [hunk 3]
```diff
@@ -126,7 +126,7 @@ class CVector : public std::vector<TData>
     CVector ( const int iNeSi ) { Init ( iNeSi ); }
     CVector ( const int iNeSi, const TData tInVa ) { Init ( iNeSi, tInVa ); }
 
-    CVector ( CVector const& ) = default;
```
Comment (pljones) [inline src/util.h:129]: See, why did clang-format consider this okay on main but want it changed here?

File: src/settings.cpp [hunk 4]
```diff
@@ -934,20 +936,20 @@ void CServerSettings::ReadSettingsFromXML ( const QDomDocument& IniXMLDocument,
...
-        }
+            else if ( GetNumericIniSet ( IniXMLDocument,
+                                         "server",
+                                         "directorytype",
+                                         static_cast<int> ( AT_NONE ),
+                                         static_cast<int> ( AT_CUSTOM ),
+                                         iValue ) )
+            {
+                directoryType = static_cast<EDirectoryType> ( iValue );
+            }
```
Comment (pljones) [inline src/settings.cpp:952]: Yup, this is wrong...

File: src/serverrpc.cpp (hunk 2 above)
Comment (pljones) [reply src/serverrpc.cpp]: It still fails without this line changing.
image

File: src/serverrpc.cpp [hunk 5]
```diff
@@ -119,7 +119,7 @@ CServerRpc::CServerRpc ( CServer* pServer, CRpcServer* pRpcServer, QObject* pare
             { "city", pServer->GetServerCity() },
             { "countryId", pServer->GetServerCountry() },
             { "welcomeMessage", pServer->GetWelcomeMessage() },
-            { "directoryServer", dsName },
+            { "directoryServer", dsName }, // TODO rename to 'directory' and update doccomment above
```
Comment (ann0see) [inline src/serverrpc.cpp:122]: I would change right here. The API can have breaking changes.

File: src/serverrpc.cpp [hunk 6]
```diff
@@ -233,7 +233,7 @@ QJsonValue CServerRpc::SerializeRegistrationStatus ( ESvrRegStatus eSvrRegStatus
         return "registered";
 
     case SRS_SERVER_LIST_FULL:
-        return "directory_server_full";
+        return "directory_server_full"; // TODO - rename to "server_list_full"
```
Comment (ann0see) [inline src/serverrpc.cpp:236]: Why not directory_(list)_full?

Comment (pljones) [reply src/serverrpc.cpp:236]: Because it's called the "server list".

Comment (ann0see) [reply src/serverrpc.cpp:236]: Then the terminology should be added to the style and tone guide: https://jamulus.io/contribute/Style-and-Tone

Comment (pljones) [reply src/serverrpc.cpp:236]: OK, see https://github.com/jamulussoftware/jamuluswebsite/pull/910.

File: src/serverrpc.cpp (hunk 5 above)
Comment (pljones) [reply src/serverrpc.cpp:122]: I'd rather do it in a separate change.  This one is explicitly _not_ meant to make functional changes.  I'll try to remember to base a new change off this with the fixes.

Comment (pljones) [reply src/serverrpc.cpp:122]: See https://github.com/pljones/jamulus/tree/bugfix/rpc-remaining-directory-server-mentions
(I don't see an easy way to compare that branch to the source of this pull request.)

**Comment by pljones** [comment]
OK, I don't think there's anything outstanding here now?

File: src/serverlist.cpp [hunk 7]
```diff
@@ -184,12 +184,13 @@ CServerListManager::CServerListManager ( const quint16  iNPortNum,
     /*
      * Init server list entry (server info for this server) with defaults.
      *
-     * The client will use the built in or custom address when retrieving the server list from a directory.
```
Comment (ann0see) [inline src/serverlist.cpp:187]: Removed on purpose?

**Review by ann0see — COMMENTED** [review]
See comment.

File: src/serverlist.cpp (hunk 7 above)
Comment (pljones) [reply src/serverlist.cpp:187]: Yes.  This block comment doesn't describe _client_ behaviour.  It's about the _server_ (whether a _directory_ or not).

File: src/settings.cpp [hunk 8]
```diff
@@ -886,7 +886,8 @@ void CServerSettings::ReadSettingsFromXML ( const QDomDocument& IniXMLDocument,
     }
 
     // to avoid multiple registrations, must do this after collecting serverinfo
-    if ( !CommandLineOptions.contains ( "--centralserver" ) && !CommandLineOptions.contains ( "--directoryserver" ) )
+    if ( !CommandLineOptions.contains ( "--centralserver" ) && !CommandLineOptions.contains ( "--directoryserver" ) &&
+         !CommandLineOptions.contains ( "--directoryaddress" ) )
```
Comment (ann0see) [inline src/settings.cpp:890]: I hope this doesn't get even more bloated in future...

File: src/main.cpp [hunk 9]
```diff
@@ -273,13 +273,13 @@ int main ( int argc, char** argv )
                                  argv,
                                  i,
                                  "--centralserver", // no short form
-                                 "--centralserver",
+                                 "--directoryserver",
```
Comment (ann0see) [inline src/main.cpp:276]: I'd add a comment that this is an alternative for compatibility reasons.

**Review by ann0see — APPROVED** [review]
I think by sight reviewing this looks ok.

File: src/settings.cpp (hunk 8 above)
Comment (pljones) [reply src/settings.cpp:890]: Well, we could step to 4.0.0 and drop a whole load of backwards compatible stuff that's almost certainly not required.

(There's backwards compatiblity breakage even within 3.x.x, of course.)

File: src/main.cpp (hunk 9 above)
Comment (pljones) [reply src/main.cpp]: Done

**Comment by pljones** [comment]
Internal changes to remove most of remaining "directory server" mentions in variable names and the like.

CHANGELOG: Rename --directoryserver to --directoryaddress

**Comment by pljones** [comment]
https://github.com/jamulussoftware/jamulus/actions/runs/4713980244/jobs/8379530170 is still failing on main (but didn't on the branch, which was up to date with main...).  Anyone understand why?

**Comment by ann0see** [comment]
@pljones I think we can drop the needs documentation flag here too.

**Pull request merged**